File: 1481-least-number-of-unique-integers-after-k-removals/1481-least-number-of-unique-integers-after-k-removals.py
"""
우선 dict에 개수 저장하기
dictNum={1:2,2:1,3:3,4:1}

(1,2),(2,1) 등으로 뽑아서 1 인덱스 기준으로 정렬
(2,1),(4,1),(1,2),(3,3)
"""

from collections import Counter


class Solution:
    def findLeastNumOfUniqueInts(self, arr: List[int], k: int) -> int:
        dictNum=Counter(arr)
        if k==0:
            return len(dictNum)
        for i,c in enumerate(sorted(dictNum.values())):
            if k>0:
                k-=c
            if k==0:
                return len(dictNum)-i-1
            elif k<0:
                return len(dictNum)-i
        return len(dictNum)
    
    

File: 1481-least-number-of-unique-integers-after-k-removals/1481-least-number-of-unique-integers-after-k-removals.cpp
#include <map>
#include <vector>

class Solution {
public:
    int findLeastNumOfUniqueInts(vector<int>& arr, int k) {
        map<int,int> counter;
        for (int num : arr){
            ++counter[num];
        }
        vector<int> counts;
        for (const auto& pair: counter){
            counts.push_back(pair.second);
        }
        sort(counts.begin(),counts.end());
        int r=counter.size();
        for (int i:counts){
            if (k >= i){
                k-=i;
                --r;
            } else {
                break;
            }
        }
        return r;
    }
};